src/libraries/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/LengthAttribute.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+
+namespace System.ComponentModel.DataAnnotations
+{
+    /// <summary>
+    ///     Specifies the minimum and maximum length of collection/string data allowed in a property.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Property | AttributeTargets.Field | AttributeTargets.Parameter, AllowMultiple = false)]
+    public class LengthAttribute : ValidationAttribute
+    {
+        [RequiresUnreferencedCode(CountPropertyHelper.RequiresUnreferencedCodeMessage)]
+        public LengthAttribute(int minimumLength, int maximumLength)
+            : base(SR.LengthAttribute_ValidationError)
+        {
+            MinimumLength = minimumLength;
+            MaximumLength = maximumLength;
+        }
+
+        /// <summary>
+        ///     Gets the minimum allowable length of the collection/string data.
+        /// </summary>
+        public int MinimumLength { get; }
+
+        /// <summary>
+        ///     Gets the maximum allowable length of the collection/string data.
+        /// </summary>
+        public int MaximumLength { get; }
+
+        /// <summary>
+        ///     Determines whether a specified object is valid. (Overrides <see cref="ValidationAttribute.IsValid(object)" />)
+        /// </summary>
+        /// <remarks>
+        ///     This method returns <c>true</c> if the <paramref name="value" /> is null.
+        ///     It is assumed the <see cref="RequiredAttribute" /> is used if the value may not be null.
+        /// </remarks>
+        /// <param name="value">The object to validate.</param>
+        /// <returns>
+        ///     <c>true</c> if the value is null or its length is between the specified minimum length and maximum length, otherwise
+        ///     <c>false</c>
+        /// </returns>
+        /// <exception cref="InvalidOperationException">
+        ///     <see cref="MinimumLength"/> is less than zero or <see cref="MaximumLength"/> is less than <see cref="MinimumLength"/>.
+        /// </exception>
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:RequiresUnreferencedCode", Justification = "The ctor is marked with RequiresUnreferencedCode.")]
+        public override bool IsValid(object? value)
+        {
+            // Check the lengths for legality
+            EnsureLegalLengths();
+
+            int length;
+            // Automatically pass if value is null. RequiredAttribute should be used to assert a value is not null.
+            if (value is null)
+            {
+                return true;
+            }
+            if (value is string str)
+            {
+                length = str.Length;
+            }
+            else if (CountPropertyHelper.TryGetCount(value, out int count))
```

Comment:
can't you use `length` directly here? 

```C#
else if (!CountPropertyHelper.TryGetCount(value, out int length))
{
    throw new InvalidCastException(SR.Format(SR.LengthAttribute_InvalidValueType, value.GetType()));
}
```

Reply:
Seems like it. I just copied the code from `MinLengthAttribute` for this method:

https://github.com/dotnet/runtime/blob/71034dd2fbbd2304fac5c24d3a9f764a3c65f781/src/libraries/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/MinLengthAttribute.cs#L60-L68

Reply:
Fixed.